products/dns/terraform.yaml

```diff
@@ -0,0 +1,53 @@
+# Copyright 2017 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Provider::Terraform::Config
+overrides: !ruby/object:Provider::ResourceOverrides
+  ManagedZone: !ruby/object:Provider::Terraform::ResourceOverride
+    examples: |
+      ```hcl
+      resource "google_dns_managed_zone" "prod" {
+        name = "prod-zone"
+        dns_name = "prod.mydomain.com."
+        description = "Production DNS zone"
+        labels = {
+          foo = "bar"
+        }
+      }
+      ```
+    properties:
+      creationTime: !ruby/object:Provider::Terraform::PropertyOverride
+        exclude: true
+      description: !ruby/object:Provider::Terraform::PropertyOverride
+        description: |
+          A textual description field. Defaults to 'Managed by Terraform'.
+        default_value: 'Managed by Terraform'
```

Comment:
We could move this to api.yaml and replace the default description to 'Managed by magic-modules' if you prefer, but that may be confusing to people using the generated modules and unaware of magic-modules are. The description field is required by the API so we need to have a default value.

Reply:
Nah, the alternative would just be to make the field Required. If it's required by the API, then we want to have that known in api.yaml (or provide a default there). I'm fine with the default instead of Required, but I also like comments in code that does things that aren't immediately intuitive :)

Reply:
Ok, I marked the `description` as required in api.yaml and kept the terraform default here for convenience.

Reply:
Changing the description to required in the API marks it as Required in Terraform and with a default value, which fails testing:
```
resource google_dns_managed_zone: description: Default cannot be set with Required
```

So either we (1) leave it as it was (not required in api.yaml), (2) we leave it required in api.yaml, override in terraform.yaml fas not required, but with a description, or (3) leave it required in api.yaml, with no default value in terraform.yaml.

I prefer (2).